File: Data Structures & Algorithms/remove-element/submission-4.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} val
     * @return {number}
     */
    removeElement(nums, val) {
        let k = nums.length;
        let i = 0;

        while (i < k) {
            if (nums[i] === val) {
                nums[i] = nums[k - 1];
                k -= 1;
            } else {
                i += 1;
            }
        }

        return k;
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-4.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        count_map = {}

        for num in nums:
            if num in count_map:
                count_map[num] += 1
            else:
                count_map[num] = 1

        buckets = []

        for i in range(len(nums) + 1):
            buckets.append([])

        for key, value in count_map.items():
            buckets[value] = [key]

        print(buckets)

        result = []

        for bucket in reversed(buckets):
            if bucket != []:
                result.append(bucket[0])

            if len(result) == k:
                return result

            